src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/CommentHelper.java

```diff
@@ -461,11 +461,63 @@ public  String getReferencedSignature(DocTree dtree) {
         return new ReferenceDocTreeVisitor<String>() {
             @Override
             public String visitReference(ReferenceTree node, Void p) {
-                return node.getSignature();
+                return normalizeSignature(node.getSignature());
             }
         }.visit(dtree, null);
     }
 
+    @SuppressWarnings("fallthrough")
+    private static String normalizeSignature(String sig) {
+        if (sig == null
```

Comment:
Consider removing this null check since we've agreed that ReferenceTree.getSignature() does not return null.

Reply:
> Consider removing this null check since we've agreed that ReferenceTree.getSignature() does not return null.

Argh... I was all the time thinking about the `Objects.requireNonNullElse` null check in `HtmlDocletWriter`, not this one. Yes, you are absolutely right, this one is superfluous... Unfortunately I already integrated the PR. I guess I can remove it in a future commit.